### DISCUSSION TITLE
Extenting Canvas for APNG file and retain transparent for original file

### DISCUSSION THREAD

**Discussion by willand-png** [ImageMagick/ImageMagick#3127] [Help]
Dear alls,

Recently, I am a newer to use ImageMagick.  Recently, I've tried to use ImageMagick to process batch of Animated PNG files for my work.  All I need to do is "extenting canvas of the APNG files to specific size".

Cos there are so many parameters in the command line, I've tried a million times and find the following statement to do so...

magick convert apng:input.apng -gravity center -background none -extent 512x512 d:\output.apng

However, the each frame in input.apng is background transparent.  Once I converted it, each frame will be overlapped with the previous one.  The result I want is like some animation that is the previous frame will be removed.

I've tried to change the above command with "-background white" and the result is like animation BUT I got no transparent for the whole APNG file.  >_<

I've searched a lot in the web, and also tried to use "-dispose" option but still not help.

It is very nice to get help from this discussion place.  Thanks for your kindly support.

Attached file for testing
[test.zip](https://github.com/ImageMagick/ImageMagick/files/5818277/test.zip)

**Comment by faran-tenovos** [comment]
Have you tried using -dispose Background?

**Comment by snibgo** [comment]
What version of IM, on what platform? Please ALWAYS give these when asking questions.

Does the problem occur if you remove "-background none -extent 512x512"?
